### ISSUE TITLE
Kevlar/Leather padding armor formulas don't make sense

### ISSUE THREAD

**Issue by ejseto** [CleverRaven/Cataclysm-DDA#11417]
In item::bash_resist() for example,

```
        k_padding = volume() * (float(get_coverage()) / 100) / 2.5;
        if (k_padding > 5 ){
            k_padding = k_padding / 2.5;
        }
```

i.e. there is a discontinuity such that k_padding = 5 is better than k_padding = 6. Should be changed to something like:

```
        k_padding = volume() * (float(get_coverage()) / 100) / 2.5;
        if (k_padding > 5 ){
            k_padding = 5 + (k_padding - 5) / 2.5;
        }
```

Similar discontinuities exists for leather and both paddings in item::cut_resist().

**Comment by Coolthulhu** [comment]
Scaling armor with coverage makes even less sense and unlike hard cap on armor, is actually harmful to balance. Good coverage items are already good, bad coverage items are already bad. No need to make it even worse.

**Comment by ejseto** [comment]
That doesn't make it worse. It makes it consistent. If you wanna just cap the padding, that's one thing, but for the additional armor to have a discontinuity such that LITERALLY 5 > 6 that makes no sense.

**Comment by KA101** [comment]
For the n+1th time, I'd only intended that formula to apply to pocketing, and specifically NOT armor, as coverage already applied (whether or not the armor took the hit, as ejseto notes).

**Comment by Coolthulhu** [comment]
I'm not blaming you specifically. Just reminding that the problem exists and is a problem.

There's no realistic reason for discontinuity in armor, yeah. I misunderstood the topic at first.
Though I think the fix should be either a hard cap, a nerf hard enough that the cap isn't reached or a scaling continuous function (square root?).

**Comment by PropaneSoup** [comment]
Looking back, that was a poor decision, and I'll take the responsibility for it. So does scaling the protection with the item's volume make sense?  I'm looking at some creating an s curve for the bonus, and it needs something to scale from. Or should it just be a flat bonus?

**Comment by KA101** [comment]
I'm thinking a fixed bonus; bulkier gear doesn't derive more protection from leather-reinforcement than lower-volume gear.

**Comment by Coolthulhu** [comment]
I'd say material thickness.

Volume doesn't make sense in many cases. For example duster has more volume than sleeveless trenchcoat, but shouldn't protect each separate bodypart more. Volume per body part could work, but would be unintuitive.

Flat bonus would need to be small, or else you'd get people armoring t-shirts. Unless you also want to add a penalty for armoring (penalty would be good - at the moment it's all free armor), having people layer armored t-shirts instead of titanium trenchcoats isn't really a solution.

Material thickness is good because it represents armor-like items already. T-shirts have it low, while hoodies have quite a bit of it. Hoodies are thick enough to support leather patches.
Though the bonus should be small, because 8 armor for 0 encumbrance is a stupidly huge number.

**Comment by PropaneSoup** [comment]
Using material thickness is going pretty well, t-shirts are getting a small bonus while jackets and boots are getting a bit more. I might have a PR up later today with some better equations. 

My plan for to counteract the 'free protection' is to increase granularity of encumbrance by 10, so adding leather could give something like +4 encumbrance, so it wouldn't do anything by itself, but it would add up between the pieces of clothing.

**Comment by Coolthulhu** [comment]
Granular encumbrance sounds OK, though it would be cool if it "spilled over" into layering penalty if it isn't enough for a full "encumbrance point".
Layer penalty is already calculated in floats, so if you simply added the penalty there, you wouldn't need to multiply all encumbrance numbers by 10.

Whichever solution you chose, inform the player about the precise penalty number in item menu. For example "encumbrance 0 (fits), layering penalty 0.3".

Semi-unrelated idea: you could reverse the order in which you use the tailoring kit. Instead of selecting the "mode" (pockets, reinforce etc.) first, you'd select the item first and then you could list what would be the exact effects of each addition rather than having players guess them.

**Comment by ejseto** [comment]
I vote fixed bonus. Scaling with volume/thickness doesn't really make sense any more than scaling with coverage. Just because something is bigger or thicker doesn't mean you can cram more kevlar into it, especially when it's supposed to not increase the encumbrance.

**Comment by Coolthulhu** [comment]
It's less about cramming more kevlar into it and more about it being capable of supporting leather/kevlar reinforcements at all. Fixed bonus with no encumbrance would mean armor t-shirts everywhere. A bit like in Nethack.

**Comment by PropaneSoup** [comment]
> list what would be the exact effects of each addition rather than having players guess them.

I have a branch improving the menu showing the player exactly what adding fur or leather is going to do, it just conflicts heavily with the wool PR that I have, so I'm waiting for that to make it in before I create a PR for it.

**Comment by KA101** [comment]
Hey, gang, material-thickness is what drives the current armor values, so adding material would increase material already present and be less-than-predictable.  I'd prefer fixed bonus, and have **no problem whatever** with adding encumbrance.

**Comment by Rivet-the-Zombie** [comment]
>adding encumbrance

This really does seem like it needs a finer granularity to the encumbrance system first, though.

**Comment by kevingranade** [comment]
Addressed by #11427, if it's still an issue though, feel free to reopen.
